src/language/tut4.ne

```diff
@@ -1,35 +1,94 @@
-
+# TUTORIAL 4
 @{%
-var semaIR = require('./semaIR.js');
+var semaIR = require('./semaIR.js'); // load additional sound engine functions
 console.log(semaIR);
 
-const moo = require("moo"); // this 'require' creates a node dependency
+const moo = require("moo"); // load tokeniser
+
+// This time, as well as sample names and whitespace, we want to match integers, and we want to introduce a separator character. Our plan is to play samples back as before, but now also to manipulate their speed. In our new language, typing:
+
+  // click>2
+
+// would play the <click> sample at double speed. */
 
 const lexer = moo.compile({
-  click:         /click/,
-  convol1:       /convol1/,
-  heart:       /heart/,
-  separator:  />/,
-  number:       /-?(?:[0-9]|[1-9][0-9]+)(?:\.[0-9]+)?(?:[eE][-+]?[0-9]+)?\b/,
+  click:        /click/,
+  convol1:      /convol1/,
+  heart:        /heart/,
+  separator:    />/,
+  number:       /-?(?:[0-9]|[1-9][0-9]+)(?:\.[0-9]+)?(?:[eE][-+]?[0-9]+)?\b/, // This is a standard regex sequence for matching integers.
```

Comment:
this regex matches not only integers but also negative, floating point, exponential number